### PR TITLE
refactor: rebase alert channel resource on the new New Relic client

### LINKED ISSUE #66: Wrong Type used in newrelic_alert_channel webhook configuration
Hi there,

Noticed that a wrong type was used for keys `payload` and `headers` in the web hook configuration

Got an error that payload and headers should be string which is not so according to the [API documentation](https://docs.newrelic.com/docs/alerts/rest-api-alerts/new-relic-alerts-rest-api/rest-api-calls-new-relic-alerts#channels-list) with example below

```
{
  "base_url": "http://test.com",
  "auth_username": "username",
  "auth_password": "password",
  "payload_type": "application/json",
  "payload": {"account_id": 1, "account_name": "account name" },
  "headers": {"header1": "test", "header2": "test"}
}
```

### Terraform Version
v0.11.10

### Affected Resource(s)
Please list the resources as a list, for example:
- newrelic_alert_channel

### Debug Output
```
2018-11-14T12:54:29.861+0400 [DEBUG] plugin.terraform-provider-newrelic_v1.2.0_x4: 2018/11/14 12:54:29 [INFO] Reading New Relic alert channel 2024412
2018/11/14 12:54:32 [ERROR] root: eval: *terraform.EvalRefresh, err: newrelic_alert_channel.this: [DEBUG] Error setting Alert Channel Configuration: &errors.errorString{s:"configuration.payload: '' expected type 'string', got unconvertible type 'map[string]interface {}'"}
2018/11/14 12:54:32 [ERROR] root: eval: *terraform.EvalSequence, err: newrelic_alert_channel.this: [DEBUG] Error setting Alert Channel Configuration: &errors.errorString{s:"configuration.payload: '' expected type 'string', got unconvertible type 'map[string]interface {}'"}
2018/11/14 12:54:32 [TRACE] [walkRefresh] Exiting eval tree: newrelic_alert_channel.this

2018/11/14 12:54:32 [DEBUG] plugin: waiting for all plugin processes to complete...
Error: Error refreshing state: 1 error(s) occurred:

2018-11-14T12:54:32.015+0400 [DEBUG] plugin.terraform-provider-newrelic_v1.2.0_x4: 2018/11/14 12:54:32 [ERR] plugin: plugin server: accept unix /var/folders/mr/6gwkzzjs4nldgm7gf2dbj6yw0000gn/T/plugin499091540: use of closed network connection
2018-11-14T12:54:32.015+0400 [DEBUG] plugin.terraform-provider-aws_v1.43.2_x4: 2018/11/14 12:54:32 [ERR] plugin: plugin server: accept unix /var/folders/mr/6gwkzzjs4nldgm7gf2dbj6yw0000gn/T/plugin205253133: use of closed network connection
* newrelic_alert_channel.this: 1 error(s) occurred:

2018-11-14T12:54:32.018+0400 [DEBUG] plugin: plugin process exited: path=/Users/oba/.terraform.d/plugins/darwin_amd64/terraform-provider-newrelic_v1.2.0_x4
2018-11-14T12:54:32.019+0400 [DEBUG] plugin: plugin process exited: path=/Users/oba/.terraform.d/plugins/darwin_amd64/terraform-provider-aws_v1.43.2_x4
* newrelic_alert_channel.this: newrelic_alert_channel.this: [DEBUG] Error setting Alert Channel Configuration: &errors.errorString{s:"configuration.payload: '' expected type 'string', got unconvertible type 'map[string]interface {}'"}
```
### Expected Behavior
`payload` and `headers` should be a map

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. create `main.tf`
```
resource "newrelic_alert_channel" "this" {
  name     = "example"
  type     = "webhook"

  configuration = {
    base_url     = "https://example.com/runbook_url"
    payload_type = "application/json"
    payload      = "${map("account_id","$ACCOUNT_ID")}"
    headers      = "${map("X-Insert-Key","my_key")}"
  }
}
```
2. `terraform apply`

### PR DESCRIPTION
**Pull request by ctrombley**
Resolves #66.

This PR swaps out the `go-newrelic` client for the new `newrelic-client-go` client.  In doing so it deprecates the `newrelic_alert_channel` resource's `configuration` attribute in favor of a new `config` block.

### CODE REVIEW

File: newrelic/provider.go [hunk 1]
```diff
@@ -8,18 +8,19 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/httpclient"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 
-	"github.com/newrelic/newrelic-client-go/pkg/config"
-	"github.com/newrelic/newrelic-client-go/pkg/infrastructure"
-	"github.com/newrelic/newrelic-client-go/pkg/synthetics"
+	"github.com/newrelic/newrelic-client-go/newrelic"
```
Comment (jthurman42) [inline newrelic/provider.go]: During the transition, should we continue to rename this import to `nr` for consistency / readability?

File: newrelic/provider.go [hunk 2]
```diff
@@ -129,31 +130,25 @@ func providerConfigure(data *schema.ResourceData, terraformVersion string) (inte
...
 		CACertFile:         data.Get("cacert_file").(string),
 	}
-	log.Println("[INFO] Initializing New Relic client")
+	log.Println("[INFO] Initializing go-newrelic client")
 
-	client, err := config.Client()
+	client, err := cfg.Client()
 	if err != nil {
-		return nil, fmt.Errorf("error initializing New Relic client: %s", err)
+		return nil, fmt.Errorf("error initializing go-newrelic client: %s", err)
```
[the remark this answers is not in the document]
Comment (jthurman42) [reply newrelic/provider.go]: Seems reasonable (fixme bot comment)

File: newrelic/resource_newrelic_alert_channel.go [hunk 3]
```diff
@@ -86,32 +72,129 @@ func resourceNewRelicAlertChannel() *schema.Resource {
...
+							Optional: true,
+						},
+						"recipients": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"region": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"US", "EU"}, false),
```
Comment (jthurman42) [inline newrelic/resource_newrelic_alert_channel.go]: Nice.   Might be a good place to add `newrelic.Region.Names` or something that returns a `[]string` of the valid regions

File: newrelic/provider.go [hunk 4]
```diff
@@ -8,18 +8,19 @@ import (
...
 const TerraformProviderProductUserAgent = "terraform-provider-newrelic"
 
 const (
-	insightsInsertURL = "https://insights-collector.newrelic.com/v1/accounts"
-	insightsQueryURL  = "https://insights-api.newrelic.com/v1/accounts"
+	apiBaseURL               = "https://api.newrelic.com/v2"
+	syntheticsAPIBaseURL     = "https://synthetics.newrelic.com/synthetics/api/v3"
+	infrastructureAPIBaseURL = "https://infra-api.newrelic.com/v2"
+	insightsInsertURL        = "https://insights-collector.newrelic.com/v1/accounts"
+	insightsQueryURL         = "https://insights-api.newrelic.com/v1/accounts"
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply newrelic/provider.go]: all the region stuff is not exported at the moment.

File: newrelic/provider.go (hunk 1 above)
Comment (ctrombley) [reply newrelic/provider.go]: 👍

File: newrelic/resource_newrelic_alert_channel.go (hunk 3 above)
Comment (ctrombley) [reply newrelic/resource_newrelic_alert_channel.go]: This argument is specific to the OpsGenie alert channel, though.  There isn't a supported "staging" environment, for instance.

File: newrelic/data_source_newrelic_alert_channel.go [hunk 5]
```diff
@@ -57,10 +157,13 @@ func dataSourceNewRelicAlertChannelRead(d *schema.ResourceData, meta interface{}
...
-	d.Set("name", channel.Name)
-	d.Set("type", channel.Type)
 	d.Set("policy_ids", channel.Links.PolicyIDs)
-	d.Set("configuration", channel.Configuration)
+
+	err = flattenAlertChannel(channel, d)
+
+	if err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply newrelic/data_source_newrelic_alert_channel.go]: 👍

**Comment by sanderblue** [comment]
> > Should we add in #292 into this refactor as well (if it's possible to change the validation without breaking things)
> 
> I think I'd like to keep it separate, should be an easy enough change.

I can make sure the #292 update is added as a fast-follow if you're good with that.

File: newrelic/data_source_newrelic_alert_channel.go [hunk 6]
```diff
@@ -57,10 +157,7 @@ func dataSourceNewRelicAlertChannelRead(d *schema.ResourceData, meta interface{}
 	}
 
 	d.SetId(strconv.Itoa(channel.ID))
```
Comment (sanderblue) [inline newrelic/data_source_newrelic_alert_channel.go]: Should we include this as part of `flattenAlertChannel`? Would put all the "Set" methods in single area.

File: newrelic/provider.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (jthurman42) [reply newrelic/provider.go]: Ah right.

File: newrelic/data_source_newrelic_alert_channel.go (hunk 6 above)
Comment (ctrombley) [reply newrelic/data_source_newrelic_alert_channel.go]: This flatten method is shared between the alert channel data source and resource. Usage of `SetID` changes based on whether or not the resource is found on the backend, so I'm not sure we can have a common treatment for that one.  `policy_ids` isn't an attribute on the resource currently.

File: newrelic/provider.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ctrombley) [reply newrelic/provider.go]: 👍

File: newrelic/data_source_newrelic_alert_channel.go (hunk 6 above)
Comment (sanderblue) [reply newrelic/data_source_newrelic_alert_channel.go]: Ah, okie dokie.

**Pull request merged**